server/base/src/main/java/org/apache/accumulo/server/tabletserver/MemoryManager.java

```diff
@@ -32,8 +33,22 @@
 
   /**
    * Initialize the memory manager.
+   *
+   * @deprecated since 2.1 use {@link #init(ServerContext)}
+   */
+  @Deprecated
+  default void init(ServerConfiguration conf) {
+    throw new IllegalStateException("Deprecated method scheduled for removal in 3.0.0");
+  }
+
+  /**
+   * Initialize the memory manager.
+   *
+   * @param context
+   *          ServerContext passed from the Tablet server
+   * @since 2.1
    */
-  void init(ServerConfiguration conf);
+  void init(ServerContext context);
```

Comment:
My only concern with this is that context is not public API any more than ServerConfiguration was. The object has already changed from a ServerConfiguration object to a ServerConfigurationFactory object (which implemented ServerConfiguration, just to make this API happy), and now ServerContext.

While it's unlikely that too many people have implemented their own MemoryManager, it is configurable, and this churn in the API isn't good for implementers. I think it would be cool if we could follow up on this PR with one that promotes this interface to SPI, and uses a stable MemoryManagerEnvironment object to inject here in place of the internal ServerContext.

Reply:
I will back out the changes to MemoryManager (which won't be hard since I separated the commits) for something better in another PR.